src/NuGet.Core/NuGet.Build.Tasks/RestoreTask.cs

```diff
@@ -242,5 +314,243 @@ public void Dispose()
         {
             _cts.Dispose();
         }
+
+#if IS_DESKTOP
+        private async Task<RestoreSummary> PerformNuGetV2RestoreAsync(Common.ILogger log, DependencyGraphSpec dgFile)
```

Comment:
I ended up copy/pasting most of this method.  Another option is to move it to `NuGet.PackageManagement` so its shared between `nuget.exe` and `RestoreTask`.  But it would have to be public so I wasn't sure what you'd prefer